rust/arrow/src/array/array_list.rs

```diff
@@ -31,12 +31,19 @@ use crate::datatypes::*;
 
 /// trait declaring an offset size, relevant for i32 vs i64 array types.
 pub trait OffsetSizeTrait: ArrowNativeType + Num + Ord + std::ops::AddAssign {
+    fn is_large() -> bool;
+
     fn prefix() -> &'static str;
 
     fn to_isize(&self) -> isize;
 }
 
 impl OffsetSizeTrait for i32 {
+    #[inline]
+    fn is_large() -> bool {
+        false
+    }
+
     fn prefix() -> &'static str {
```

Comment:
maybe we won't need `prefix` anymore with the new `is_large`.

Reply:
We might still need it, we also use it for formatting in `Display`

Reply:
I think that we can drop it, yes. We can merge StringOffset, BinaryOffset and OffsetTrait in a single Trait with this, but I wanted to leave it to another PR.

Reply:
Here is one way to remove `prefix` that does not go as far as @jorgecarleitao  suggests to collapse the traits... https://github.com/apache/arrow/pull/9508/commits/8e68e05160a56f9feb64438e4dcd739a67a884e1